clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeliverExecutor.scala

```diff
@@ -0,0 +1,15 @@
+package org.broadinstitute.clio.client.dispatch
+
+import org.broadinstitute.clio.client.commands.DeliverCommand
+import org.broadinstitute.clio.transfer.model.DeliverableIndex
+
+/**
+  * Special-purpose CLP for delivering crams to FireCloud workspaces.
+  *
+  * Wraps the move CLP with extra IO / upsert logic:
+  *
+  *   1. Writes the cram md5 value to file at the target path
+  *   2. Records the workspace name in the metadata for the delivered cram
+  */
+abstract class DeliverExecutor[DI <: DeliverableIndex](deliverCommand: DeliverCommand[DI])
```

Comment:
What does having this inbetween MoveExecutor and Deliver(Arrays|WgsCram)Executor buy us?

Reply:
It lets us, in `ComandDispatch`, not have to special-case anything.

Reply:
oh argh no it doesn't cause it's abstract. I'll fiddle with this one too.